src/field/pokemon.ts

```diff
@@ -3523,28 +3542,52 @@ export abstract class Pokemon extends Phaser.GameObjects.Container {
     return false;
   }
 
-  getTag<T extends BattlerTag = BattlerTag>(...tagTypes: BattlerTagType[]): T | undefined {
+  /**
+   * @param tagTypes - The {@linkcode BattlerTagType}s to retrieve
+   * @returns The first {@linkcode BattlerTag} found of the specified type(s), or `undefined` if none is found
+   */
+  public getTag<T extends BattlerTag = BattlerTag>(...tagTypes: BattlerTagType[]): T | undefined {
     return this.summonData.tags.find((t) => tagTypes.includes(t.tagType)) as T | undefined;
   }
 
   /**
    * Helper function to check if a Pokemon has any of the input tag types.
    * @param tagTypes - The battler tag types to search for
-   * @returns `true` if the Pokemon has at least one of the battler tags, `false` otherwise
+   * @returns Whether the Pokemon has at least one of the input battler tags
    */
   public hasTag(...tagTypes: BattlerTagType[]): boolean {
+    if (tagTypes.length === 0) {
+      console.warn("`Pokemon#hasTag` called with no parameters!");
+      console.trace();
+      return false;
+    }
     return this.summonData.tags.some((t) => tagTypes.includes(t.tagType));
   }
 
-  findTag<T extends BattlerTag = BattlerTag>(tagFilter: (tag: BattlerTag) => boolean): T | undefined {
-    return this.summonData.tags.find((t) => tagFilter(t)) as T | undefined;
+  /**
+   * Filters the {@linkcode BattlerTag}s attached to the pokemon and returns the first result
+   * @param tagFilter - The filter to apply to the tag list
+   * @returns The first {@linkcode BattlerTag} found, or `undefined` if none is found
+   */
+  public findTag<T extends BattlerTag = BattlerTag>(tagFilter: (tag: T) => boolean): T | undefined {
+    return this.summonData.tags.find((t: T) => tagFilter(t)) as T | undefined;
```

Comment:
```suggestion
    return this.summonData.tags.find(tagFilter) as T | undefined;
```
I also wonder if `tagFilter` could be made into a type guard to avoid the casting, but I feel like it wouldn't be so simple...

Reply:
Typeguard like `is T`?